packages/SwingSet/src/kernel/vatManager/externalStore.js

```diff
@@ -0,0 +1,56 @@
+// @ts-check
+
+import {
+  makeStore,
+  makeWeakStore,
+  makeClosureExternalStoreMaker,
+} from '@agoric/store';
+
+// TODO: Give this parameters so that it can interact with the vat's liveslots
+// and send syscalls for actual external stores.
+export function makeVatExternalStoreMaker(vatId) {
+  /** @type {Store<number, InstanceStore>} */
+  const storeIdToInstanceStore = makeStore(`vat ${vatId} storeId`);
+  // console.warn('making external store');
+  return makeClosureExternalStoreMaker(_instanceHook => ({
+    getInstanceStore(storeId) {
+      // TODO: Load the hydrate store instance for VATID.STOREID.
+      return storeIdToInstanceStore.get(storeId);
+    },
+    makeInstanceStore(storeId, instanceKind) {
+      // console.warn(`makeInstanceStore for ${instanceKind}`);
+      // TODO: Use the vat-specific store instead of an in-memory store.
+      const store = makeStore(`${instanceKind} ids`);
+      /** @type {InstanceStore} */
+      const istore = {
+        get(instanceId) {
+          // TODO: Fetch and deserialize the VATID.STOREID.INSTANCEID closure
+          // data.
+          const data = store.get(instanceId);
+
+          // Note that we prevent a memory leak here.  We rely on the fact that
+          // unless we call a `instanceHook` method, there is exactly one `get`
+          // per id.
+          store.delete(instanceId);
```

Comment:
When I explain such a destructive read, I often use the word "take" for the operation. Note that I am not suggestion that we add a `take` operation to the code. I am also *not requesting* that it be added to the comments. Just offering in case useful